### ISSUE TITLE
Ask for help when reproduce your paper result

### ISSUE THREAD

**Issue by jijiangli-swu-worker** [CGLab-GIST/target-aware-denoising#1]
Hi, I want to reproduce your paper's addon in mistuba3 optimization schedule, after rendering I want do with your excellent denoising work, but I am not familiar with docker as your readme mentioned, I only want to use your work as addon in mitsuba3 gradient-based optimizer, so if possible, I am here to ask you for help me how to start with it and use it directly in mitsuba3, any reply would be appreciate! Also, thanks for your contribution on directly denoising in PBIR.

**Comment by JeongminGu** [comment]
Hi.
I'm not sure what problem you have when running the provided code, but I guess you may have issues with docker.
You can test our code by installing PyTorch and custom operators without docker. 
If you have any other problem, please let me know.

**Comment by jijiangli-swu-worker** [comment]
Hi,
I currently reproduce your paper's result, very well-written paper for proposed method, especially for the intention of this work in introduction. But I'm confused with some key points about denoising Monte Carlo Rendering, so I'm here to ask for your to help me better understand this task cause I'm not a specialist in denoising.
## About low spp issue in PBIR
As you have mentioned that if using infinite spp in PBIR which will cause f(pi) with the noise-free image in rendering. But actually for low spp even if before backpropagation using a denoiser to get a $\hat{f}$ the bias may also seem exist too much between denoised image to target when testing. As shown in Fig 1. your rendering image and other method rendered image during test stage all seemed  noisy not like a higher spp rendered. So I wonder how noisy this will happen when use low spp in PBIR problem or does using low spp to recover a very good rendering image be a very hard issue in PBIR? I'm not sure about this. If this was hard, does it mean that even I have a very good denoiser like your method proposed, I can not even get a very good result like using high spp?
## about formular (7) to (9)
I understand the very high idea was to utilize GT make the noisy image to be less noise before compute the gradient during training. This to guarantee even with low spp your method can get the low biased gradient with respect to scene params, actually it does the way like your paper said. But I have something not really understood.  If you would, could tell me how to set bandwidth and windowSize for different scenes. I want to change them cause I found that the denoiser(linear regression for the $\hat{f}$) actually calculate for right way in training. But sometimes this will "learn" more about scene parameters so that in test time, without using a GT denoiser, the result are full of strange color area. I mean the denoiser may **utilize** the GT to make a noisy rendered like GT and will "take off" some parameters correctness learned in denoiser. Or does this would be the common issue in low spp rendering?
## about fireflies noise in rendering
when I use high spp render a noisy image, I met the fireflies noise in result, and this are not disappearing even I increase very high spp. If you could, do you have any suggestion about how to deal with such noise in PBIR before auto differentiation?
![image](https://github.com/CGLab-GIST/target-aware-denoising/assets/175003723/4261279a-56c1-4008-bf5b-218692d72dd1)
I'm sorry to ask your too many questions. For you actually the expert in denoising, If possible, I wish to learn some experience from your constructive suggestion. Any reply would be appreciate!

**Comment by JeongminGu** [comment]
Hi,
1) It is a common problem in Monte Carlo rendering that MC denoiser has residual variance or lots of bias when using low sample counts (high variance). Nevertheless, adapting denoiser at low sample counts is helpful, as tested in our paper.
2) I recommend the paper below, which provides a detailed explanation of adaptive bandwidth selection.
* Recent Advances in Adaptive Sampling and Reconstruction for Monte Carlo Rendering. Zwicker et al. 2015. 
3) I'm not sure how fireflies affect optimizing scene parameters that you want to infer. If you use path tracing, it is very hard to remove them without any post-process, such as firefly removers. Also, you can use other integrators (e.g., bidirectional path tracing) in a differentiable rendering framework.

**Comment by jijiangli-swu-worker** [comment]
Thanks for your patient replies and great advice! I am really inspired by your excellent work!
